runtime/src/main/java/org/corfudb/runtime/view/LayoutManagementView.java

```diff
@@ -110,6 +110,38 @@ public void bootstrapNewNode(String endpoint)
         log.info("bootstrapNewNode: New node {} bootstrapped.", endpoint);
     }
 
+    /**
+     * Removes a node from the layout
+     * @param currentLayout the layout to remove the node from
+     * @param endpoint the node to remove
+     */
+    public void removeNode(Layout currentLayout, String endpoint) throws QuorumUnreachableException,
+            CloneNotSupportedException,
+            LayoutModificationException, OutrankedException {
+
+        LayoutBuilder builder = new LayoutBuilder(currentLayout);
+        Layout newLayout = builder.removeLayoutServer(endpoint)
+                .removeSequencerServer(endpoint)
+                .removeLogunitServer(endpoint)
+                .removeUnResponsiveServer(endpoint)
+                .setEpoch(currentLayout.getEpoch() + 1)
+                .build();
+
+        // Seal after constructing the layout, so that the system
+        // isn't blocked if the builder throws an exception
+        currentLayout.setRuntime(runtime);
+        sealEpoch(currentLayout);
+
+        newLayout.setRuntime(runtime);
+        attemptConsensus(newLayout);
+
+        try {
+            reconfigureSequencerServers(currentLayout, newLayout, false);
+        } catch (InterruptedException | ExecutionException e) {
+            log.error("removeNode: Bootstrapping sequencer failed due to exception : ", e);
```

Comment:
Don't catch InterruptedException. Rethrow UnrecoverableCorfuInterruptedError

Reply:
Also rethrow any other specific exceptions, or throw an UnrecoverableCorfuError if you really didn't expect that exception there.

Reply:
I'm not sure what you are saying. Both exceptions are specific. If I dont catch the exception, how do I re-throw it?

Reply:
Just because they are specific doesn't mean you can swallow them. Either wrap them in unrecoverableError or handle the exception. Handling InterruptedException means you need to stop looping, call Thread.currentThread().interrupt() and clean up any state. None of which you do.  Throwing UnrecoverableCorfuInterruptedException() (and calling Thread.currentThread.interrupt() is one way to take care of it.

Reply:
I see how swallowing the interrupt is wrong. I just don't see how the UnrecoverableCorfuError can't be swallowed. Are we trying to "throw new RuntimeException(e)", but we have this special exception so that sonar cube doesn't complain ?

Reply:
Well, throwing new RuntimeException would be one way to fix the problem, but UnrecoverableCorfuError is there, and it exists so other catch (Exception e) {} blocks that still lurk in the code base can't catch and swallow errors as well

Reply:
Ok, fixed the interrupted exception with UnrecoverableCorfuInterruptedError.

I tried wrapping some of the checked exceptions with UnrecoverableCorfuError, but it turned out to be elaborate and messy. I will not do them in this change. Some of the exceptions incorrectly extend from Exception, where they should really extend from RuntimeException. This taints the code with try/catch/throws. I think that is the fundamental issue that should be addressed, along with removing catch-all statements. Also, in my opinion using UnrecoverableCorfuError to solve a review problem indirectly is not effective. If you recognize that you should throw a UnrecoverableCorfuError, then you would also be able to recognize that the exception should be a runtime exception.